dali/base/dautils.cpp

```diff
@@ -3083,3 +3113,262 @@ bool traceAllTransactions(bool on)
     return ret;
 }
 
+const char* LockDataFieldNames[] = { "@XPath", "@EPCount", "@EP", "@connId", "@sessId", "@mode", "@timeLocked" };
+
+extern da_decl const char* getLockDataFieldName(LockDataField feild)
+{
+    return LockDataFieldNames[feild];
+}
+
+void CLockDataHelper::serializeLockData(CheckedCriticalSection &crit, unsigned critTimeout, const char *xpath, ConnectionInfoMap &connectionInfo, CMessageBuffer &mb)
+{
+    unsigned epCount=0;
+    MemoryBuffer locks;
+    UInt64Array keys;
+    {
+        CHECKEDCRITICALBLOCK(crit, critTimeout);
+        HashIterator iter(connectionInfo);
+        ForEach(iter)
+        {
+            IMapping &imap = iter.query();
+            LockData *lD = connectionInfo.mapToValue(&imap);
+            keys.append(* ((ConnectionId *) imap.getKey()));
+            locks.append(sizeof(LockData), lD);
+            epCount++;
+        }
+    }
+    mb.append(xpath);
+    mb.append(epCount);
+    if (epCount < 1)
+        return;
+
+    ForEachItemIn(k, keys)
+    {
+        LockData lD;
+        memcpy(&lD, ((const byte *)locks.toByteArray())+k*sizeof(LockData), sizeof(LockData));
+
+        StringBuffer sessEpStr;
+        mb.append(querySessionManager().getClientProcessEndpoint(lD.sessId, sessEpStr).str());
+        mb.append(lD.sessId);
+        mb.append(keys.item(k));
+        mb.append(lD.mode);
+        mb.append(lD.timeLockObtained);
+    }
```

Comment:
Not clear to me, why this functionaliy (serializing content of CLockInfo) is in not in CLockInfo..
looks unclean passing a crit in for example..
Would be better in a  : 
CLockInfo::serialize(MemoryBuffer &mb)